diff --git a/email_confirm.module b/email_confirm.module
index a397c3a..0d591fb 100644
--- a/email_confirm.module
+++ b/email_confirm.module
@@ -174,7 +174,7 @@ function email_confirm_user_presave(&$edit, $account, $category) {
     // Set a temporary session variable to indicate that the email was changed. Is used in hook_exit
     // to clear out a Drupal message set by the user module after user_save() is called.
     $_SESSION['email_changed'] = TRUE;
-    email_confirm_build_mail($edit);
+    email_confirm_build_mail($edit, $account);
     $edit['data']['email_confirm']['pending_email'] = $edit['mail'];
     $edit['data']['email_confirm']['expiration_time'] = (REQUEST_TIME + 86400);
     module_invoke_all('email_confirm', 'email change', $account->uid, $account->mail, $edit['mail']);
@@ -310,12 +310,11 @@ function email_confirm_mail($key, &$message, $params) {
  * Build and send out the confirmation email to the user's
  * current and proposed new email address.
  */
-function email_confirm_build_mail($edit) {
-  global $user;
+function email_confirm_build_mail($edit, $account) {
   $params = array();
-  $params['account'] = $user;
+  $params['account'] = $account;
 
-  $params['context']['url'] = email_confirm_confirmation_email_url($edit);
+  $params['context']['url'] = email_confirm_confirmation_email_url($edit, $account);
 
   if (module_exists('smtp') && variable_get('smtp_from', '') != '') {
     $default_from = $smtp_from;
@@ -332,8 +331,8 @@ function email_confirm_build_mail($edit) {
     $params['headers']['Bcc'] = $bcc;
   }
 
-  if ($message['result'] = drupal_mail('email_confirm', 'confirmation_email', $edit['mail'], user_preferred_language($user), $params, $from)) {
-    drupal_mail('email_confirm', 'confirmation_original_email', $user->mail, user_preferred_language($user), $params, $from);
+  if ($message['result'] = drupal_mail('email_confirm', 'confirmation_email', $edit['mail'], user_preferred_language($account), $params, $from)) {
+    drupal_mail('email_confirm', 'confirmation_original_email', $account->mail, user_preferred_language($account), $params, $from);
     drupal_set_message(t('A confirmation email has been sent to your new email address. You must follow the link provided in that email within 24 hours in order to confirm the change to your account email address.'));
   }
 }
@@ -415,13 +414,12 @@ function email_confirm_mail_tokens(&$replacements, $data, $options) {
 /**
  * Generate a URL to confirm an email address change request.
  */
-function email_confirm_confirmation_email_url($edit) {
-  global $user;
+function email_confirm_confirmation_email_url($edit, $account) {
   $timestamp = REQUEST_TIME;
-  $pass = isset($edit['pass']) ? md5($edit['pass']) : $user->pass;
+  $pass = isset($edit['pass']) ? md5($edit['pass']) : $account->pass;
   $hash = email_confirm_user_email_rehash($pass, $edit['mail']);
 
-  return url('user/change-mail/' . $user->uid . '/' . $timestamp . '/' . $edit['mail'] . '/' . $hash, array('absolute' => TRUE));
+  return url('user/change-mail/' . $account->uid . '/' . $timestamp . '/' . $edit['mail'] . '/' . $hash, array('absolute' => TRUE));
 }
 
 /**
